drivers/input/i8042prt/pnp.c

```diff
@@ -263,6 +264,44 @@ i8042DetectMouse(
     INFO_(I8042PRT, "Mouse not detected\n");
 }
 
+static void
+i8042DetectMouseType(
+    IN PPORT_DEVICE_EXTENSION DeviceExtension)
+{
+    int chkcnt = 0;
+    UCHAR Value;
+    /* Set the mouse to streaming mode, and disable error reporting (as per https://isdaman.com/alsos/hardware/mouse/ps2interface.htm) CORE-6901, CORE-12663, CORE-12434 */
+    i8042IsrWritePort(DeviceExtension, MOU_SET_STREAM_MODE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, MOU_NO_ERROR_REPORT, CTRL_WRITE_MOUSE);
+	
+    /* Initialize Intellimouse or Intellimouse Explorer compatible mice (This fixes wheels) */
+    i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, 0xC8, CTRL_WRITE_MOUSE);
+
+    i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, 0x64, CTRL_WRITE_MOUSE);
+	
+    i8042IsrWritePort(DeviceExtension, MOU_DEF_SAMPLE_RATE, CTRL_WRITE_MOUSE);
+    i8042IsrWritePort(DeviceExtension, 0x50, CTRL_WRITE_MOUSE);
+
+    i8042IsrWritePort(DeviceExtension, MOU_READ_DEV_TYPE, CTRL_WRITE_MOUSE);
+
+    while(chkcnt < 10) {
```

Comment:
Fix code style please.

Reply:
Also you can use a for-loop instead.

Reply:
Is a for - loop more efficient? 
(or anything else)

Reply:
Actually, I just tried to use a for-loop, and it broke the mouse logic

Reply:
> Is a for-loop more efficient?
> (or anything else)

No, but it would be more readable.

Reply:
How about that... :slightly_smiling_face: 

```
int chkcnt = 10;

while (chkcnt--)
{
    ...
}
```